models/packed_dataset.py

```diff
@@ -0,0 +1,235 @@
+# Very loosely inspired by indexed_dataset in Fairseq, Megatron
+# https://github.com/NVIDIA/Megatron-LM/blob/main/megatron/data/indexed_dataset.py
+
```

Comment:
nit: Not a model, maybe add a `/data`

Reply:
yea I agree and considered this, but left it all crammed into the same dir